product_multiple_barcodes/wizard/multiply_barcode_wizard.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 VentorTech OU
+# Part of Ventor modules. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class MultiplyBarcodeWizard(models.TransientModel):
+
+    _name = 'multiply.barcode.wizard'
+
+    name = fields.Char(
+        string='New Barcode',
+        required=True,
+    )
+
+    check_previous_barcode = fields.Boolean(
+        string='Remember previous barcode in "Additional Barcode field"',
+    )
+
+    def update_barcode(self):
+        model_name = self.env.context['active_model']
+        if model_name == 'product.product':
+            product = self.env['product.product'].browse(self.env.context['active_id'])
+        if model_name == 'product.template':
+            product = self.env['product.template'].browse(self.env.context['active_id'])
+
+        if self.check_previous_barcode:
+            barcode = product.barcode
+            if barcode:
+                product_barcode_multi = self.env['product.barcode.multi'].create({
+                    'name': barcode,
+                    'product_id': product.id,
```

Comment:
It is not correct. Product_id is reference to product.product. and you are putting there reference to product.template (in case clicking button from template).

Sometimes product.tempate and product.product I'd can the same. But it is not the rule. They are usually different . So as result you will attach barcodes to incorrect product 

I see that in line below, you just write to o2m filed barcode_ids directly. Maybe even no need to put here any value?